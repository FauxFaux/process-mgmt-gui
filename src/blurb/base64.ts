import * as base64 from '@protobufjs/base64';

export function unB64(encoded: string): Uint8Array {
  const deWeb = encoded.replace(/-/g, '+').replace(/_/g, '/');
  const padded = deWeb.padEnd(
    deWeb.length + ((4 - (deWeb.length % 4)) % 4),
    '=',
  );
  const len = base64.length(padded);
  const buffer = new Uint8Array(len);
  base64.decode(padded, buffer, 0);
  return buffer;
}

export function enB64(arr: Uint8Array): string {
  return base64
    .encode(arr, 0, arr.length)
    .replace(/=/g, '')
    .replace(/\+/g, '-')
    .replace(/\//g, '_');
}
