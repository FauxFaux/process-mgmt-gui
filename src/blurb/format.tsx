export const handleColours = (text: string) => {
  const bits = text.split(/(.*?)\[color=(\d+,\d+,\d+)]([^[]+)\[\/color]/g);
  if (bits.length === 1) {
    return <>{text}</>;
  }
  console.log(bits);
  const chunks = 2;
  const parts = [];
  for (let i = 0; i < bits.length; i += chunks) {
    if (i === bits.length - 1) {
      parts.push(<>{bits[i]}</>);
      break;
    }

    const [colour, content] = bits.slice(i, i + chunks);
    if (colour) {
      parts.push(<span style={`color: rgb(${colour})`}>{content}</span>);
    } else {
      parts.push(<>{content}</>);
    }
  }

  return <>{parts}</>;
};
