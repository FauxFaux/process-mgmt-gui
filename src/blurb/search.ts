export const regexpOrNot = (s: string): { test: (s: string) => boolean } => {
  try {
    const regex = new RegExp(s, 'i');
    return {
      test: (s) => regex.test(s),
    };
  } catch (e) {
    console.warn(e);
    return {
      test: (s) => s.includes(s),
    };
  }
};
