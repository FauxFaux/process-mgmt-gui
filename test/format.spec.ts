import { it, expect } from '@jest/globals';
import renderToString from 'preact-render-to-string';

import { handleColours } from '../src/blurb/format';

it('handles some colour cases', () => {
  const f = (text: string) => renderToString(handleColours(text));
  expect(f('plain string')).toMatchSnapshot();
  expect(
    f('one[color=1,2,3]two[/color]three[color=10,20,30]four[/color]five'),
  ).toMatchSnapshot();
});
