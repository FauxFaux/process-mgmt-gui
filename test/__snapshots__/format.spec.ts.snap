// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handles some colour cases 1`] = `"plain string"`;

exports[`handles some colour cases 2`] = `"content<span style="color: rgb(1,2,3)">two</span>content<span style="color: rgb(10,20,30)">four</span>five"`;
